frontend/public/components/cluster-service-classes.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+
+import { ColHead, DetailsPage, List, ListHeader, ListPage, ResourceRow } from './factory';
+import { SectionHeading, detailsPage, navFactory, ResourceLink, ResourceSummary } from './utils';
+// eslint-disable-next-line no-unused-vars
+import { K8sResourceKind, K8sResourceKindReference } from '../module/k8s';
+import * as _ from 'lodash-es';
+
+const ServiceClassesReference: K8sResourceKindReference = 'ClusterServiceClass';
+
+const normalizeIconClass = (iconClass) => {
+  return _.startsWith(iconClass, 'icon-') ? `font-icon ${iconClass}` : iconClass;
+};
+
+export const ServiceClassIcon: React.SFC<ServiceClassIconProps> = ({iconClass, imageUrl, imageClass}) => {
```

Comment:
This should take in the service class resource itself and read the iconClass / imageUrl from it so the logic is in one place